multichain-testing/test/fast-usdc/fast-usdc.test.ts

```diff
@@ -0,0 +1,291 @@
+
+  const instances = await vstorageClient.queryData(
+    'published.agoricNames.instance',
+  );
+  const instance = fromEntries(instances)[contractName];
+
+  // accept oracle operator invitations
+  await Promise.all(
+    oracleWds.map(makeDoOffer).map((doOffer, i) =>
+      doOffer({
+        id: toOracleOfferId(i),
+        invitationSpec: {
+          source: 'purse',
+          instance,
+          description: 'oracle operator invitation', // TODO export/import INVITATION_MAKERS_DESC
+        },
+        proposal: {},
+      }),
+    ),
+  );
+
+  for (const name of keys(oracleMnemonics)) {
+    const addr = wallets[name];
+    await t.notThrowsAsync(() =>
+      retryUntilCondition(
+        () => vstorageClient.queryData(`published.wallet.${addr}.current`),
+        ({ offerToUsedInvitation }) => {
+          return offerToUsedInvitation[0][0] === `${name}-accept`;
+        },
+        `${name} invitation used`,
+      ),
+    );
+  }
+});
+
+test.serial('lp deposits', async t => {
+  const { lpUser, retryUntilCondition, vstorageClient } = t.context;
+
+  const lpDoOffer = makeDoOffer(lpUser);
+  const brands = await vstorageClient.queryData('published.agoricNames.brand');
+  const { USDC } = Object.fromEntries(brands);
+
+  const usdcGive = make(USDC, 10_000_000n);
+
+  const { shareWorth: currShareWorth } = await vstorageClient.queryData(
+    `published.${contractName}.poolMetrics`,
+  );
+
+  await lpDoOffer({
+    id: `lp-deposit-${Date.now()}`,
+    invitationSpec: {
+      source: 'agoricContract',
+      instancePath: [contractName],
+      callPipe: [['makeDepositInvitation']],
+    },
+    proposal: {
+      give: { USDC: usdcGive },
+      want: { PoolShare: divideBy(usdcGive, currShareWorth) },
+    },
+  });
+
+  await t.notThrowsAsync(() =>
+    retryUntilCondition(
+      () => vstorageClient.queryData(`published.${contractName}.poolMetrics`),
+      ({ shareWorth }) =>
+        !isGTE(currShareWorth.numerator, shareWorth.numerator),
+      'share worth numerator increases from deposit',
+    ),
+  );
+});
+
+test.serial('advance and settlement', async t => {
+  const {
+    nobleTools,
+    nobleAgoricChannelId,
+    oracleWds,
+    retryUntilCondition,
+    useChain,
+    usdcOnOsmosis,
+    vstorageClient,
+  } = t.context;
+
+  // EUD wallet on osmosis
+  const eudWallet = await createWallet(useChain('osmosis').chain.bech32_prefix);
+  const eudAddress = (await eudWallet.getAccounts())[0].address;
+
+  // parameterize agoric address
+  const { settlementAccount } = await vstorageClient.queryData(
+    `published.${contractName}`,
+  );
+  // TODO #10614 use bech32 encoding
+  const recipientAddress = `${settlementAccount}?EUD=${eudAddress}`;
+  t.log('recipientAddress', recipientAddress);
+
+  // register forwarding address on noble
+  const txRes = nobleTools.registerForwardingAcct(
+    nobleAgoricChannelId,
+    recipientAddress,
+  );
```

Comment:
hm. is this step in the diagrams in our current spec?
IOU a check.

Reply:
I am a little hazy - but iirc the UI's backend service and/or OCW does this step

Reply:
ah yes... it's there. arrow 4.

```mermaid
sequenceDiagram
    title Fast USDC Transaction Process
    participant NEA as Noble Express app<br/>[Browser]
    participant NC as Noble Chain<br/>[Noble]

rect rgb(240, 248, 255)
        NEA-->>NC: Register NFA
end
```